### ISSUE TITLE
Exporting Figures: Cell Counts

### ISSUE THREAD

**Issue by timobermeier** [jupyter/nbconvert#1316]
Hi all,

I was generating some graphs and ran into a problem with the cell counts when there are markdown cells in the notebook.  I'm converting my notebook into an html file with the following command:

jupyter nbconvert --to html_toc --execute test.ipynb --output "test"

The notebook basically looks like this: 

![notebook_example](https://user-images.githubusercontent.com/2004807/86746346-e882b780-c032-11ea-9a8d-a1ff4e6ba4de.png)

And the generated output files (in the folder "test_files") are:

![generated_output](https://user-images.githubusercontent.com/2004807/86746952-55964d00-c033-11ea-8e6a-ba27b54171d6.png)

The issue is that the files numbers don't coincide with the cell counts in the notebook. The second graph gets labelled as "6_1" even though it's generated in cell 3. It seems like nbconvert counts the markdown cells, which are not included in the count in the Jupyter notebook. 

In large notebooks, this makes it hard to find the right outputs among the generated files. I didn't find a discussion about this so far - is there currently a good way to get around this problem?

Best
Tim

**Comment by MSeal** [comment]
What's the `html_toc` template defined as? That's not a built-in template and the image outputs for standard `html` are embded pngs so there's no filename for them.

**Comment by timobermeier** [comment]
html_toc comes from this extension:

https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/nbextensions/toc2/README.html

I just tried a couple of the built-in templates. The export with "asciidoc" leads to the same issue (the second graph is labelled 6_1):

![nbconvert_ascii](https://user-images.githubusercontent.com/2004807/86896793-711a5a00-c0fe-11ea-9ac6-e5aa3eb84cbe.png)

**Comment by t-makaro** [comment]
This is done by the ```extractOutputPreprocessor``` to ensure that the output files have unique names. We could change it there, but if a kernel is restarted then two cells can have the same execution count which could cause a problem.

**Comment by MSeal** [comment]
> This is done by the extractOutputPreprocessor to ensure that the output files have unique names. We could change it there, but if a kernel is restarted then two cells can have the same execution count which could cause a problem.

I don't think we have any nbconvert processes that support kernel restarts mid-execution. But if we did image extraction from markdown and needed to save an image file we wouldn't have a unique index to reuse if we leaned on the kernel execution counts. This is solvable as well in that we can use a different naming convention if we get to that.

Technically the cell execution counts seen in the notebook from a fresh Run All are not required to be the numbers you are seeing. I'm somewhat inclined to leave the cell index naming convention to keep it matching the original document rather than the kernel count state after clean execution, but I could be swayed either way.

**Comment by timobermeier** [comment]
Maybe one could add an option to allow switching between these modes? My impression is that matching the kernel counts would be more convenient in some cases - in my other notebook, there are ~100 graphs and it takes a while to go through all the files and find the right ones (for example for using it in a presentation).

**Comment by MSeal** [comment]
I'd be ok if someone were to add it as an option flag to extractOutputPreprocessor if @t-makaro is also ok with it -- I probably don't have bandwidth to get to this for quite a while but I can review a PR if someone else makes the change.

**Comment by t-makaro** [comment]
> I don't think we have any nbconvert processes that support kernel restarts mid-execution.

I was just thinking if someone ran the notebook from Jupyter Lab or the classic interface and didn't convert with ```--execute```. I don't think that the extractOutputPreprocessor touches markdown cells, but I could be wrong.

I think that the easiest solution would be to make the naming format of the output something like ```"output_{cell#}_{execution_count_if_available}_{#of_the_output}"```. Just adding the one extra number to the file name.

It should be possible to add this to the current config. https://github.com/jupyter/nbconvert/blob/a4bf9b152e069b88eeb303c64c898829d8b497b1/nbconvert/preprocessors/extractoutput.py#L49

**Comment by timobermeier** [comment]
Yes, that sounds like a good solution! Would somebody among you have time to do this or should I look into it myself? I'm not familiar with the nbconvert codes, so I'm not sure if I'd manage.
